code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/dao/impl/ServiceExposeMapDaoImpl.java

```diff
@@ -154,4 +103,52 @@ public void updateEnvironmentName(Environment env) {
                                         CommonStatesConstants.REMOVING))))
                 .fetchInto(ServiceExposeMapRecord.class);
     }
+
+    @Override
+    public List<Service> collectSidekickServices(Service initialService, Map<String, String> initialSvcLabels) {
+        List<Service> servicesToCollect = new ArrayList<>();
+        if (initialSvcLabels == null) {
+            initialSvcLabels = sdService.getServiceLabels(initialService);
+        }
+        String initialSvcSidekickLabel = initialSvcLabels.
+                get(ServiceDiscoveryConstants.LABEL_SERVICE_SIDEKICK) == null ? null : initialSvcLabels
+                .get(ServiceDiscoveryConstants.LABEL_SERVICE_SIDEKICK);
```

Comment:
jeez :) originally it was something else, then did the refactoring and it doesn't make sense any more